File: backend/src/main/java/br/com/flagplatform/common/enums/RoundType.java
package br.com.flagplatform.common.enums;

import lombok.Getter;

@Getter
public enum RoundType implements PersistableEnum {

    REGULAR("REGULAR", "Regular"),
    PLAYOFFS("PLAYOFFS", "Playoffs");

    private final String code;
    private final String description;

    RoundType(String code, String description) {
        this.code = code;
        this.description = description;
    }

}

File: frontend/packages/domain/lib/src/enums/round_type.dart
enum RoundType {
  regular,
  playoffs;

  static RoundType fromJson(String value) => switch (value) {
        'REGULAR' => RoundType.regular,
        'PLAYOFFS' => RoundType.playoffs,
        _ => throw FormatException('Tipo de rodada desconhecido: $value'),
      };

  String toJson() => switch (this) {
        RoundType.regular => 'REGULAR',
        RoundType.playoffs => 'PLAYOFFS',
      };
}
